src/zarr/core/metadata/v2.py

```diff
@@ -300,7 +315,27 @@ def parse_metadata(data: ArrayV2Metadata) -> ArrayV2Metadata:
     return data
 
 
-def parse_fill_value(fill_value: object, dtype: np.dtype[Any]) -> Any:
+def parse_structured_fill_value(fill_value: Any, dtype: np.dtype[Any]) -> Any:
+    """Handle structured dtype/fill value pairs"""
+    try:
+        if isinstance(fill_value, list):
+            fill_value = tuple(fill_value)
```

Comment:
I think it would be clearer in this function to just return all the values in the branches, intead of assigning to `fill_value` and then immediately returning it after the try/except statement